### DISCUSSION TITLE
how to get rate limit for the "like tweet" endpoint?

### DISCUSSION THREAD

**Discussion by MarufRien1** [plhery/node-twitter-api-v2#277] [Q&A] [answered]
I am trying to get the rate limit for the "like tweet" endpoint.

```js
import { TwitterApi } from 'twitter-api-v2'
import { TwitterApiRateLimitPlugin } from '@twitter-api-v2/plugin-rate-limit'

const rateLimitPlugin = new TwitterApiRateLimitPlugin()
const client = new TwitterApi(yourKeys, { plugins: [rateLimitPlugin] })

// ...make requests...
await client.v2.like('1112', '2220');
// ...

const currentRateLimitForMe = await rateLimitPlugin.v2.getRateLimit('users/1234/likes') // 123 is my twitter id

console.log(currentRateLimitForMe) // result : undefined  
```
I didn't find anything in the docs that would help me.

I think I am doing something wrong here ---> `await rateLimitPlugin.v2.getRateLimit('users/1234/likes')`

I have tried :  users/:loggedUserId/likes, users/likes, users/loggedUserId/likes

**Comment by alkihis** [comment, marked as answer]
Hi, URL is the same as available in twitter doc: users/:id/likes
See https://developer.twitter.com/en/docs/twitter-api/tweets/likes/api-reference/post-users-id-likes

**Reply by MarufRien1** [reply]
@alkihis thank you for your response ♥

yes. I have  used us--> users/:id/likes .. but the result is always undefined 

my code:
![code](https://user-images.githubusercontent.com/74663078/166691399-89a51953-504a-407d-a6e7-600f5811939f.png)

the result: 
![image](https://user-images.githubusercontent.com/74663078/166691630-fea3b305-4fb4-4fae-b556-ec80207a5b42.png)
@

**Reply by alkihis** [reply]
users/:id is litteral, don't put your id in the string, use ":id"

**Reply by MarufRien1** [reply]
yes. I have tried using "users/:id/likes" as a string but it also shows undefined.

my code :
![code](https://user-images.githubusercontent.com/74663078/166721111-c189dccc-ab49-4faf-bb46-ceb3bc45ef0d.png)



the result:
![image](https://user-images.githubusercontent.com/74663078/166720379-84a94e76-08da-4cec-9bf8-326aa1e32057.png)

**Reply by alkihis** [reply]
You forgot the `.v2`.
Valid code: `rateLimitPlugin.v2.getRateLimit('users/:id/likes', 'POST')`
